contracts/InnerOwnedSet.sol

```diff
@@ -0,0 +1,127 @@
+//!
+//!     http://www.apache.org/licenses/LICENSE-2.0
+//!
+//! Unless required by applicable law or agreed to in writing, software
+//! distributed under the License is distributed on an "AS IS" BASIS,
+//! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//! See the License for the specific language governing permissions and
+//! limitations under the License.
+//!
+//! Original code taken from https://github.com/paritytech/contracts
+
+pragma solidity ^0.4.23;
+
+import "./interfaces/Owned.sol";
+import "./interfaces/RelaySet.sol";
+
+// Owner can add or remove validators.
+
+contract InnerOwnedSet is Owned, InnerSet {
+	// EVENTS
+	event Report(address indexed reporter, address indexed reported, bool indexed malicious);
+	event InitiateChange(bytes32 indexed _parent_hash, address[] _new_set);
+	event ChangeFinalized(address[] current_set);
+
+	// System address, used by the block sealer.
+	uint public recentBlocks = 20;
+
+	modifier is_validator(address _someone) {
+		if (pendingStatus[_someone].isIn) { _; }
+	}
+
+	modifier is_pending(address _someone) {
+		require(pendingStatus[_someone].isIn);
+		_;
+	}
+
+	modifier is_not_pending(address _someone) {
+		require(!pendingStatus[_someone].isIn);
+		_;
+	}
+
+	modifier is_recent(uint _blockNumber) {
+		require(block.number <= _blockNumber + recentBlocks);
+		_;
+	}
+
+	struct AddressStatus {
+		bool isIn;
+		uint index;
+	}
+
+	// Current list of addresses entitled to participate in the consensus.
+	address[] validators;
+	address[] pending;
+	mapping(address => AddressStatus) pendingStatus;
+
+	constructor(address[] _initial) public {
+		pending = _initial;
+		for (uint i = 0; i < _initial.length - 1; i++) {
+			pendingStatus[_initial[i]].isIn = true;
+			pendingStatus[_initial[i]].index = i;
+		}
+		validators = pending;
+	}
+
+	// Called to determine the current set of validators.
+	function getValidators() constant public returns (address[]) {
+		return validators;
+	}
+
+	function getPending() constant public returns (address[]) {
+		return pending;
+	}
+
+	// Log desire to change the current list.
+	function initiateChange() private {
+		outerSet.initiateChange(block.blockhash(block.number - 1), getPending());
+		emit InitiateChange(block.blockhash(block.number - 1), getPending());
+	}
+
+	function finalizeChange() public only_outer {
+		validators = pending;
+		emit ChangeFinalized(validators);
+	}
+
+	// OWNER FUNCTIONS
+
+	// Add a validator.
+	function addValidator(address _validator) public only_owner is_not_pending(_validator) {
+		pendingStatus[_validator].isIn = true;
+		pendingStatus[_validator].index = pending.length;
+		pending.push(_validator);
+		initiateChange();
+	}
+
+	// Remove a validator.
+	function removeValidator(address _validator) public only_owner is_pending(_validator) {
+		pending[pendingStatus[_validator].index] = pending[pending.length - 1];
+		delete pending[pending.length - 1];
+		pending.length--;
```

Comment:
as per above. reducing the length has the effect of dropping the last element in the array